report.php

```diff
@@ -138,14 +122,32 @@
 $where = implode(' AND ', $where);
 
 $sql = "SELECT x.id, x.eventname, u.firstname, u.lastname, x.contextid, x.timecreated, $extraselect
-          FROM {$basetable} x
+          FROM {{$basetable}} x
      LEFT JOIN {user} u
             ON u.id = x.userid
          WHERE $where";
+
+// Resend elements.
+$canresenderrors = $fromform = $mform->get_data() && !empty($fromform->resend) && $fromform->resend == XAPI_REPORT_RESEND_TRUE && $canmanageerrors;
+
+if ($canresenderrors) {
+    $eventids = array_keys($DB->get_records_sql($sql, $params));
+
+    if (!empty($eventids)) {
+        $mover = new \logstore_xapi\log\moveback($eventids, $id);
+        if ($mover->execute()) {
+            $notifications[] = new notification(get_string('resendevents:success', 'logstore_xapi'), notification::NOTIFY_SUCCESS);
+        } else {
+            $notifications[] = new notification(get_string('resendevents:failed', 'logstore_xapi'), notification::NOTIFY_ERROR);
+        }
+    }
+}
+
+// Collect events to create view.
 $results = $DB->get_records_sql($sql, $params, $page*$perpage, $perpage);
 
 $sql = "SELECT COUNT(x.id)
-          FROM {$basetable} x
+          FROM {{$basetable}} x
```

Comment:
Is this a typo? {{$basetable}}

Reply:
Nope one for concat, one for moodle table handler.